hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimeGenerator.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.timeline;
+
+public interface TimeGenerator {
+
+  void lock();
+
+  /**
+   * Generate a globally monotonically increasing timestamp. All implements must ensure
+   * 1. any caller can compute a timestamp T that is guaranteed to be greater than
+   * any timestamp T' if T' finished being generated before T started being generated.
+   * 2. This guarantee holds across all different callers.
+   *
+   * @param shouldLock Whether to hold a new lock inside this method. Some callers(like
+   * completion time generation) already holds lock, whereas here doesn't need to lock again.
+   *
+   * @return Current true time as millis.
+   */
+  long currentTimeMillis(boolean shouldLock);
+
```

Comment:
We already have `lock` and `unlock` so do we still need the `shouldLock` flag?

Reply:
refactor the interface to wrap the lock inside.